docs: Update Email link validation note for allowed schemes and anchor links (#837)
mautic/user-documentation: docs/channels/emails.rst

.. vale off

> When you save an Email, Mautic validates every link it contains. If a link's ``href`` uses an invalid format - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error such as 'The email contains an invalid URL: ://example.com'. A link is valid when its ``href`` uses one of these seven schemes: ``http``, ``https``, ``ftp``, ``ftps``, ``mailto``, ``tel``, or ``sms``, is an anchor link that starts with ``#``, or is a Mautic token such as ``{unsubscribe_url}``.

Reviewer: The last sentence is quite hard to read. We can use unordered list instead.
 
```suggestion
When you save an Email, Mautic validates every link it contains. If a link's ``href`` uses an invalid format - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error such as 'The email contains an invalid URL: ://example.com'.

A link is valid when its ``href``:

* Uses one of these seven schemes: ``http``, ``https``, ``ftp``, ``ftps``, ``mailto``, ``tel``, or ``sms``
* Is an anchor link that starts with ``#``
* Is a Mautic token such as ``{unsubscribe_url}``
```

> When you save an Email, Mautic validates every link it contains. If a link's ``href`` uses an invalid format - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error such as 'The email contains an invalid URL: ://example.com'. A link is valid when its ``href`` uses one of these seven schemes (``http``, ``https``, ``ftp``, ``ftps``, ``mailto``, ``tel``, or ``sms``); is an anchor link that starts with ``#``; or is a Mautic token such as ``{unsubscribe_url}``.

Reviewer: ```suggestion
When you save an Email, Mautic validates every link it contains. If a link's ``href`` uses an invalid format - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error such as 'The email contains an invalid URL: ://example.com'. A link is valid when its ``href`` uses one of these seven schemes: ``http``, ``https``, ``ftp``, ``ftps``, ``mailto``, ``tel``, or ``sms``, is an anchor link that starts with ``#``, or is a Mautic token such as ``{unsubscribe_url}``.
```

> When you save an Email, Mautic checks every link it contains. If a link's ``href`` is malformed - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error beneath the editor, such as 'The email contains an invalid URL: ://example.com'. This prevents broken links from going out and disrupting email delivery. A link is valid when its ``href`` uses a scheme such as ``http://``, ``https://``, ``mailto:``, or ``tel:``, is an anchor link that starts with ``#``, or is a Mautic token such as ``{unsubscribe_url}``. Fix any flagged link, then save the Email again.

Author: Broadened the list of valid link forms per [PR #16508](https://github.com/mautic/mautic/pull/16508). Its new `hasValidScheme()` method in `ValidEmailLinksValidator.php` accepts any URL beginning with `#` or with a non-empty scheme returned by `parse_url(..., PHP_URL_SCHEME)`, so `tel:`, `mailto:`, and anchor links now save; the earlier note only allowed absolute `http(s)://` URLs, `mailto:`, or tokens. The `Email/form.html.twig` change surfaces the 'invalid URL' error beneath the editor, which is why the note says the error appears beneath the editor.

Source: https://github.com/mautic/mautic/pull/16508

> When you save an Email, Mautic checks every link it contains. If a link's ``href`` is malformed - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error beneath the editor, such as 'The email contains an invalid URL: ://example.com'. This prevents broken links from going out and disrupting email delivery. A link is valid when its ``href`` uses a scheme such as ``http://``, ``https://``, ``mailto:``, or ``tel:``, is an anchor link that starts with ``#``, or is a Mautic token such as ``{unsubscribe_url}``. Fix any flagged link, then save the Email again.

Reviewer: @promptless-for-oss please address the suggestions.

```suggestion
When you save an Email, Mautic validates every link it contains. If a link's ``href`` uses an invalid format - for example ``://example.com``, which has no scheme - Mautic blocks the save and shows an error beneath the editor, such as 'The email contains an invalid URL: ://example.com'. This prevents broken links from going out and disrupting email delivery. A link is valid when its ``href`` uses a scheme such as ``http://``, ``https://``, ``mailto:``, or ``tel:``, is an anchor link that starts with ``#``, or is a Mautic token such as ``{unsubscribe_url}``. Fix any flagged link, then save the Email again.
```
